src/main/java/pl/edu/icm/pl/mxrdr/extension/workflow/XdsImagesPatternStep.java

```diff
@@ -0,0 +1,100 @@
+package pl.edu.icm.pl.mxrdr.extension.workflow;
+
+import edu.harvard.iq.dataverse.dataset.datasetversion.DatasetVersionServiceBean;
+import edu.harvard.iq.dataverse.workflow.WorkflowContext;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Computes file name pattern to pass to <code>generate_XDS.INP</code> script.
+ */
+class XdsImagesPatternStep extends XdsWorkflowStep {
+
+    static final String STEP_ID = "xds-images-pattern";
+
+    /**
+     * Input parameter containing {@value #FILE_NAMES_SEPARATOR} separated list of file names to compute pattern for.
+     */
+    static final String FILE_NAMES_PARAM_NAME = "fileNames";
+
+    /**
+     * Values separator for {@value #FILE_NAMES_PARAM_NAME} input parameter.
+     */
+    static final String FILE_NAMES_SEPARATOR = ";";
+
+    private final List<String> fileNames;
+
+    public XdsImagesPatternStep(DatasetVersionServiceBean datasetVersions, Map<String, String> inputParams) {
+        super(datasetVersions);
+        this.fileNames = asList(inputParams.getOrDefault(FILE_NAMES_PARAM_NAME, "").split(FILE_NAMES_SEPARATOR));
```

Comment:
Maybe a stupid question, but can we be sure that we do not split some filename containing a semicolon accidentally? Or  at least we state it in docs that such a name is prohibited (I think that Windows and linux allow such a character in the filename)?

Reply:
We assume the file names are produced by a process following a strict naming convention and are a result of some other automated process. The algorithm can be hacked with semicolons, but then dataset author wouldn't get the analytics they want. Also, calculated pattern will be visible in one of workflow execution artifacts to allow for dateset corrections. Hence it seems such risk is relatively low and acceptable.